src/lib/apiUtils.js

```diff
@@ -571,55 +649,30 @@ export async function transfer(request, blockchain) {
       return _promptFail("transfer", request.id, 'No toSend', reject);
     }
 
+    let targetID;
     let targetAccount = request.payload.params.to;
+    if (!targetAccount.includes("1.2.")) {
+        let targetAccountContents;
+        try {
+            targetAccountContents = await this.getAccount(targetAccount);
+        } catch (error) {
+            console.log(error);
+            return reject();
+        }
+        
+        targetID = targetAccountContents.id;
+    } else {
+        targetID = targetAccount;
+    }
+
     let blockedAccounts;
     try {
-        blockedAccounts = await blockchain.getBlockedAccounts(targetAccount);
+        blockedAccounts = await blockchain.getBlockedAccounts();
```

Comment:
Fetching it again? This seems very excessive, and not performance optimized. 

We need to find a way to fetch this once and cache it, or at least only fetch it once per request.

Reply:
In the getBlockedAccounts function it checks if it's stored in _blockedAccounts and if so just returns it, it won't retry the blockchain lookup.

Reply:
Ah, thx